src/platform/macos/cgl/surface.rs

```diff
@@ -76,6 +77,38 @@ impl Debug for SurfaceTexture {
     }
 }
 
+fn surface_bind_to_gl_texture(surface: &IOSurfaceRef, width: i32, height: i32, has_alpha: bool) {
+    const BGRA: GLenum = 0x80E1;
+    const RGBA: GLenum = 0x1908;
+    const RGB: GLenum = 0x1907;
+    const TEXTURE_RECTANGLE_ARB: GLenum = 0x84F5;
+    const UNSIGNED_INT_8_8_8_8_REV: GLenum = 0x8367;
+
+    unsafe {
+        let context = CGLGetCurrentContext();
+        let gl_error = CGLTexImageIOSurface2D(
+            context,
+            TEXTURE_RECTANGLE_ARB,
+            if has_alpha {
+                RGBA as GLenum
+            } else {
+                RGB as GLenum
+            },
+            width,
+            height,
+            BGRA as GLenum,
+            UNSIGNED_INT_8_8_8_8_REV,
+            surface as *const IOSurfaceRef as cgl::IOSurfaceRef,
+            0,
+        );
+
+        if gl_error != kCGLNoError {
+            let error_msg = CStr::from_ptr(CGLErrorString(gl_error));
+            panic!("{}", error_msg.to_string_lossy());
+        }
+    }
+}
+
```

Comment:
This was previously provided in [`io_surface::IOSurface::bind_to_gl_texture`](https://docs.rs/io-surface/0.16.0/io_surface/struct.IOSurface.html#method.bind_to_gl_texture), but it really mixes two domains (IOSurface and CGL), and requires that we link both of those, so I didn't want to expose it in `objc2-io-surface`.